src/combinators/mod.rs

```diff
@@ -28,6 +28,27 @@ pub trait BackToTheBuffer: Write {
     fn reserve_write_use<Tmp, Gen: Fn(Self) -> GenResult<(Self, Tmp)>, Before: Fn(Self, Tmp) -> GenResult<Self>>(self, reserved: usize, gen: Gen, before: Before) -> GenResult<Self> where Self: Sized;
 }
 
+pub trait Seek: Write + io::Seek {}
```

Comment:
Why do we need an additional trait here?

Reply:
Forgot to add a comment.
It's there because we have no guarantee that std will never impl Seek on slices so implementation collide

Reply:
But like this we can't use it e.g. on a `File` :)

Reply:
You can use it on some wrapper around File.

Reply:
(Yes, that suboptimal, but I don't see any simpler solution here, apart dropping the impl on slices)

Reply:
We could assume that `std` never adds the `Seek` impl. It seems unlikely enough to me.

Fine with me either way, I'm not serializing directly to a `File` or similar ;)

Reply:
I'm totally fine with that, if you can find a way to tell this to rustc ;)

I'm not the one complaining about this possibility here :)

Reply:
Oh! Never mind then :)